app/graphql/foreman_graphql_schema.rb

```diff
@@ -20,14 +20,22 @@ def self.object_from_id(id, query_ctx)
     return unless id.present?
 
     _, type_name, item_id = Foreman::GlobalId.decode(id)
-    model_class = type_name.safe_constantize
+    type_class = "::Types::#{type_name}".safe_constantize
+
+    model_class = type_class&.model_class
 
     return unless model_class
 
     RecordLoader.for(model_class).load(item_id.to_i)
   end
 
-  def self.resolve_type(_, obj, _)
-    types[obj.class.name]
+  def self.resolve_type(_, object, _)
+    klass = object.class
+    if klass.respond_to?(:graphql_type) && klass.graphql_type
+      type = klass.graphql_type
+      return type.call if type.respond_to?(:call)
+      return type.safe_constantize
+    end
+    types[klass.name.demodulize]
```

Comment:
```
Host::Managed.name.demodulize => "Managed"
```
I'm not sure, is this the expected result in this case?

Reply:
It also does not work for `Subnet::Ipv4`. However it works for `ForemanPlugin::SomeModel`, that's why I chose `demodulize`. We can simply set `graphql_type` for all namespaced models in core to mitigate this.
But I'm open to suggestions.

Reply:
I can change this back to `obj.class.name`. All namespaced plugins would need to set `graphql_type` then, which is fine imho. WDYT, @tbrisker?